nautobot/extras/tests/example_jobs/test_profiling.py

```diff
@@ -0,0 +1,24 @@
+import pstats
+
+from nautobot.extras.jobs import Job
+
+
+class TestProfilingJob(Job):
+    """
+    Job to have profiling tested.
+    """
+
+    description = "Test profiling"
+
+    def run(self, data, commit):
+        """
+        Job function.
+        """
+
+        self.log_success(obj=None, message="Profiling test.")
+
+        return []
+
+    def post_run(self):
+        # This serves as the 'assert' for the test - it shouldn't fail
+        pstats.Stats(f"/tmp/{self.job_result.id}.pstats")
```

Comment:
`post_run` runs in a try/except block that doesn't reraise so this will not fail the test unless you check the JobResult status

Reply:
Thanks, I added a check for that.